momepy/preprocessing.py

```diff
@@ -1101,3 +1105,453 @@ def roundabout_simplification(
     output = _ext_lines_to_center(edges, incoming_all, idx_drop)
 
     return output
+
+
+def consolidate_intersections(
+    graph,
+    tolerance=30,
+    rebuild_graph=True,
+    rebuild_edges_method="spider",
+    directed=None,
+    x_col="x",
+    y_col="y",
+    edge_from_col="from",
+    edge_to_col="to",
+):
+    """
+    Consolidate close street intersections into a single node, collapsing short edges.
+
+    If rebuild_graph is True, new edges are drawn according to rebuild_edges_method which is one of:
+
+    1. Extension reconstruction:
+        Edges are linearly extended from original endpoints until the new nodes. This method preserves
+        most faithfully the network geometry.
+    2. Spider-web reconstruction:
+        Edges are cropped within a buffer of the new endpoints and linearly extended from there. This
+        method improves upon linear reconstruction by mantaining, when possible, network planarity.
+    3. Euclidean reconstruction:
+        Edges are ignored and new edges are built as straightlines between new origin and new
+        destination. This method ignores geometry, but efficiently preserves adjacency.
+
+    If rebuild_graph is False, graph is returned with consolidated nodes but without reconstructed
+    edges i.e. graph is intentionally disconnected.
+
+    Graph must be configured so that
+
+    1. All nodes have attributes determining their x and y coordinates;
+    2. All edges have attributes determining their origin, destination, and geometry.
+
+    Parameters
+    ----------
+    graph : Networkx.MultiGraph or Networkx.MultiDiGraph
+    tolerance : float
+        distance in network units below which nodes will be consolidated
+    rebuild_graph: Boolean
+    rebuild_edges_method: string
+        'extend' or 'spider' or 'euclidean'
+    directed: Boolean or None
+        consider the graph a MultiDiGraph if True or MultiGraph if False, and
+        if None infer from the passed object type
+    x_col, y_col: string
+        node attribute with the valid coordinate
+    edge_from_col, edge_to_col: string
+        edge attribute with the valid origin/destination node id
+
+    Returns
+    ----------
+    Networkx.MultiGraph or Networkx.MultiDiGraph
```

Comment:
What's the name of this variable?

Reply:
That doesn't really matter, does it? It may be not even be assigned to any variable.